db/fixtures/ae_datastore/ManageIQ/Cloud/VM/Retirement/StateMachines/Methods.class/__methods__/amazon_check_pre_retirement.rb

```diff
@@ -5,21 +5,30 @@
 
 # Get vm from root object
 vm = $evm.root['vm']
+if vm.nil?
+  $evm.log('info', "VM not found, cannot retire.")
+  $evm.root['ae_result'] = 'error'
+  exit MIQ_ABORT
+end
+ems = vm.ext_management_system
+if ems.nil?
+  $evm.log('info', "VM:<#{vm.name}> has no provider.")
+  $evm.root['ae_result'] = 'ok'
+  exit MIQ_OK
+end
+power_state = vm.attributes['power_state']
 
-if vm
-  power_state = vm.attributes['power_state']
-  ems = vm.ext_management_system
-  $evm.log('info', "Instance:<#{vm.name}> on EMS:<#{ems.try(:name)} has Power State:<#{power_state}>")
-
-  # If VM is powered off or this instance is running on an instance store
-  if %w(off suspended terminated).include?(power_state) || vm.hardware.root_device_type == "instance_store"
-    # Bump State
-    $evm.root['ae_result'] = 'ok'
-  elsif power_state == "never"
-    # If never then this VM is a template so exit the retirement state machine
-    $evm.root['ae_result'] = 'error'
-  else
-    $evm.root['ae_result']     = 'retry'
-    $evm.root['ae_retry_interval'] = '60.seconds'
-  end
+$evm.log('info', "Instance: <#{vm.name}> on EMS: <#{ems.try(:name)} has Power State: <#{power_state}>")
```

Comment:
@billfitzgerald0120 
we don't need the try since ems has already been checked to be not nil
ems.try(:name) could be just ems.name